pkg/generate/app/pipeline.go

```diff
@@ -75,8 +112,41 @@ func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy
 	}, nil
 }
 
+// NewImagePipeline creates a new pipeline with components that are not expected
+// to be built.
+func (pb *pipelineBuilder) NewImagePipeline(from string, resolvedMatch *ComponentMatch) (*Pipeline, error) {
+	input, err := InputImageFromMatch(resolvedMatch)
+	if err != nil {
+		return nil, fmt.Errorf("can't include %q: %v", from, err)
+	}
+
+	name, err := pb.nameGenerator.Generate(input)
+	if err != nil {
+		return nil, err
+	}
+	input.ObjectName = name
+
+	return &Pipeline{
+		Name:  name,
+		From:  from,
+		Image: input,
+	}, nil
+}
+
+// Pipeline holds components.
+type Pipeline struct {
+	Name string
+	From string
+
+	InputImage *ImageRef
+	Build      *BuildRef
+	Image      *ImageRef
+	Deployment *DeploymentConfigRef
+	Labels     map[string]string
+}
+
 // NeedsDeployment sets the pipeline for deployment.
-func (p *Pipeline) NeedsDeployment(env Environment, labels map[string]string, name string) error {
+func (p *Pipeline) NeedsDeployment(name string, env Environment, labels map[string]string) error {
```

Comment:
Just readability. I realized we can actually get rid of "name" and always use `p.Name`.